kart/working_copy.py

```diff
@@ -462,3 +463,38 @@ def _check_for_unsupported_structural_changes(
                 "Sorry, this type of filter is not yet supported when there are structural changes."
                 f" Unfilterable structural changes are affecting:\n{unsupported_filters}"
             )
+
+
+def handle_working_copy_tree_mismatch(wc_type_name, actual_tree_id, expected_tree_id):
+    actual_tree_id = f"tree {actual_tree_id}" if actual_tree_id else "the empty tree"
+    expected_tree_id = (
+        f"tree {expected_tree_id}" if expected_tree_id else "the empty tree"
+    )
+
+    message = [
+        f"The {wc_type_name} working copy appears to be out of sync with the repository.",
+        f"The working copy's own records show it is tracking {actual_tree_id};",
+        f"based on the repository it should be tracking {expected_tree_id}.",
+        "The most straight-forward fix is generally to recreate the working copy from the ground up (losing any uncommitted changes in the process.)",
+    ]
+
+    if get_input_mode() != InputMode.INTERACTIVE:
+        message.append("To do so, try the following command:")
+        message.append("kart create-workingcopy --delete-existing --discard-changes")
+        raise BadStateError("\n".join(message), exit_code=BAD_WORKING_COPY_STATE)
+
+    click.echo("\n".join(message))
+    click.echo()
+    if not click.confirm(
+        "Do you want to recreate the working copy from the ground up?"
```

Comment:
```suggestion
        "Do you want to recreate the working copy?"
```

I'd bet that expression doesn't make sense for non-native English speakers and it doesn't really add anything